asv/plugin_manager.py

```diff
@@ -23,28 +24,37 @@ def __init__(self):
 
     def load_plugins(self, package):
         prefix = package.__name__ + "."
-        for module_finder, name, ispkg in pkgutil.iter_modules(
-            package.__path__, prefix
-        ):
+        for module_finder, name, ispkg in pkgutil.iter_modules(package.__path__, prefix):
             try:
-                __import__(name)
-                mod = sys.modules[name]
+                mod = importlib.import_module(name)
                 self.init_plugin(mod)
                 self._plugins.append(mod)
             except ModuleNotFoundError as err:
                 log.error(f"Couldn't load {name} because\n{err}")
                 continue  # Couldn't find mamba
 
+    def _load_plugin_by_name(self, name):
+        prefix = plugins.__name__ + "."
+        for module_finder, module_name, ispkg in pkgutil.iter_modules(plugins.__path__, prefix):
+            if name in module_name:
+                mod = importlib.import_module(module_name)
+                return mod
+        return None
+
     def import_plugin(self, name):
         extended = False
         if name.startswith("."):
             extended = True
             sys.path.insert(0, ".")
             name = name[1:]
         try:
-            mod = __import__(name, {}, {}, [], level=0)
-            self.init_plugin(mod)
-            self._plugins.append(mod)
+            if extended:
+                mod = importlib.import_module(name)
+            else:
+                mod = self._load_plugin_by_name(name)
+            if mod:
+                self.init_plugin(mod)
+                self._plugins.append(mod)
```

Comment:
This also fixes what I noticed in #1283, that `asv` plugins are not actually loaded correctly. Now they are.